confgenerator/filter/internal/ast/ast.go

```diff
@@ -0,0 +1,439 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ast
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/filter/internal/generated/token"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+)
+
+type Attrib interface{}
+
+type Member []string
```

Comment:
`Member` sounds like a generic concept. Could we change it to something like `FieldPath`, `TargetField` etc. for better readability?

Reply:
Sure, will change.

Reply:
I don't think naming this `FieldPath` is a good idea; it's either a field path or a literal. The name `Member` came from our public syntax: https://google.aip.dev/assets/misc/ebnf-filtering.txt

```
# Member expressions are either value or DOT qualified field references.
#
# Example: `expr.type_map.1.type`
member
    : value {DOT field}
    ;
```

If you really want to rename it, please leave an explicit comment that "This represents `member` from the filter BNF, and represents either a value or a dotted field path."

Reply:
`field` or `target` is also OK for readability benefits. Adding an explicit comment is fine.

Reply:
Changed to `target` and added an explicit comment.